listenbrainz/webserver/static/js/src/RecentListens.test.tsx

```diff
@@ -1111,3 +1111,120 @@ describe("Pagination", () => {
     });
   });
 });
+
+describe("handleClickNewerRecommendations", () => {
+  it("don't do anything if already on newest/first page i.e. offset === 0", async () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+
+    const recs = recentListensProps.listens.slice(0, 25);
+    wrapper.setState({ offset: 0, listens: recs });
+
+    await instance.handleClickNewerRecommendations();
+
+    expect(wrapper.state("offset")).toEqual(0);
+    expect(wrapper.state("listens")).toEqual(recs);
+  });
+
+  it("go to the previous page and update listens if offset !== 0", async () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+
+    wrapper.setState({
+      offset: 50,
+      listens: recentListensProps.listens.slice(50, 53),
+    });
+
+    await instance.handleClickNewerRecommendations();
+
+    expect(wrapper.state("offset")).toEqual(25);
+    expect(wrapper.state("listens").length).toEqual(25);
+  });
+});
+
+describe("handleClickNewestRecommendations", () => {
+  it("don't do anything if already on newest/first page i.e. offset === 0", async () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+
+    const recs = recentListensProps.listens.slice(0, 25);
+    wrapper.setState({ offset: 0, listens: recs });
+
+    await instance.handleClickNewestRecommendations();
+
+    expect(wrapper.state("offset")).toEqual(0);
+    expect(wrapper.state("listens")).toEqual(recs);
+  });
+
+  it("move to the newest/first page if offset !== 0", async () => {
+    const wrapper = shallow<RecentListens>(<RecentListens {...props} />);
+    const instance = wrapper.instance();
+
+    wrapper.setState({
+      offset: 50,
+      listens: recentListensProps.listens.slice(50, 53),
+    });
+
+    await instance.handleClickNewestRecommendations();
+
+    expect(wrapper.state("offset")).toEqual(0);
+    expect(wrapper.state("listens").length).toEqual(25);
```

Comment:
I am only checking the length of returned array and not the contents since `array.slice` of the same array at different places is giving different results. 

I think the `listens` array is being modified somewhere in the code, or may be there is some other reason, I am not sure.